contracts/truefi2/LoanToken2.sol

```diff
@@ -374,17 +370,25 @@ contract LoanToken2 is ILoanToken2, ERC20 {
         require(!isRepaid(), "LoanToken2: cannot default a repaid loan");
         require(start.add(term).add(creditOracle.gracePeriod()) <= block.timestamp, "LoanToken2: Loan cannot be defaulted yet");
         status = Status.Defaulted;
+        debtToken = loanFactory.createDebtToken(pool, borrower, debt.sub(repaid()));
+
+        uint256 poolBalance = balanceOf(lender);
+        uint256 debtShare = debtToken.totalSupply().mul(poolBalance).div(debt);
+        debtToken.approve(address(pool), debtShare);
+        pool.addDebt(debtToken, debtShare);
 
-        emit Defaulted(_balance());
+        emit Defaulted(debtToken, debt.sub(repaid()));
     }
 
-    /**
-     * @dev Liquidate the loan if it has defaulted
-     */
-    function liquidate() external override onlyDefaulted onlyLiquidator {
-        status = Status.Liquidated;
+    function claimDebtToken(address loanHolder) public onlyDefaulted {
```

Comment:
We can't really test this function because we have no pool.exit() anymore. I thought it's worth adding in case loans will be distributed in the future.

Reply:
Yeah, you make a good point, especially since our minimal proxies can't upgrade LoanToken2s that have already been deployed.

However, we should also prevent repeated claimDebtToken() calls on the same LoanToken2 balance. What's the best way to achieve this?
1. Burn the loanHolder's LoanToken2 balance and also subtract from the loan balance?
2. Add another mapping to track whether a lender has already claimed their debtToken balance?

If this becomes complicated to implement, I propose we don't allow outside loan claims for now. We can always introduce it in a follow-up PR.

Reply:
The safer solution would be to remove it and add later if we need to. For now I'll assume that agency holds all loan tokens